drv/riscv-plic-server/build.rs

```diff
@@ -0,0 +1,173 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use indexmap::IndexMap;
+use serde::Deserialize;
+use std::collections::{BTreeMap, HashMap};
+use std::env;
+use std::fs::File;
+use std::io::Write;
+use std::path::PathBuf;
+
+task_config::task_config! {
+    ints: &'static [u16],
+    tasks: &'static [&'static str],
+    notification: &'static [u32],
+    pbits: u8,
+}
+
+#[derive(Clone, Debug, Deserialize)]
+#[serde(rename_all = "kebab-case", deny_unknown_fields)]
+#[allow(dead_code)]
+struct Peripheral {
+    pub address: u32,
+    pub size: u32,
+    #[serde(default)]
+    pub interrupts: BTreeMap<String, u32>,
+}
```

Comment:
This structure seems to be a copy of the `Peripheral` structure in build/xtask. I think we should avoid duplicating structures if possible. Maybe as a separate change this can be moved to a common place?

Reply:
Agreed, I'll look into it for the future.